cost-analyzer/templates/aggregator-statefulset.yaml

```diff
@@ -0,0 +1,189 @@
+{{- if and (not .Values.agent) (not .Values.cloudAgent) (.Values.kubecostAggregator) }}
+{{- if .Values.kubecostAggregator.enabled }}
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "aggregator.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "aggregator.commonLabels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.kubecostAggregator.replicas }}
+  serviceName: {{ template "aggregator.serviceName" . }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: aggregator
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app: aggregator
+  volumeClaimTemplates:
+  - metadata:
+      name: persistent-configs
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      storageClassName: {{ .Values.kubecostAggregator.persistentConfigsStorage.storageClass }}
+      resources:
+        requests:
+          storage: {{ .Values.kubecostAggregator.persistentConfigsStorage.storageRequest }}
+  - metadata:
+      name: aggregator-storage
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      storageClassName: {{ .Values.kubecostAggregator.aggregatorStorage.storageClass }}
+      resources:
+        requests:
+          storage:  {{ .Values.kubecostAggregator.aggregatorStorage.storageRequest }}
+  {{- if .Values.kubecostAggregator.aggregatorDbStorage }}
+  - metadata:
+      name: aggregator-db-storage
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      storageClassName: {{ .Values.kubecostAggregator.aggregatorDbStorage.storageClass }}
+      resources:
+        requests:
+          storage:  {{ .Values.kubecostAggregator.aggregatorDbStorage.storageRequest }}
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: aggregator
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app: aggregator
+    spec:
+      restartPolicy: Always
+      {{- if .Values.kubecostAggregator.securityContext }}
+      securityContext:
+          {{- toYaml .Values.kubecostAggregator.securityContext | nindent 10 }}
+      {{ end }}
+      serviceAccountName: {{ template "aggregator.serviceAccountName" . }}
+      volumes:
+        {{- $etlBackupBucketSecret := "" }}
+        {{- if .Values.kubecostModel.federatedStorageConfigSecret }}
+            {{- $etlBackupBucketSecret = .Values.kubecostModel.federatedStorageConfigSecret }}
+        {{- end }}
+        {{- if $etlBackupBucketSecret }}
+        - name: bucket-config
+          secret:
+           defaultMode: 420
+           secretName: {{ $etlBackupBucketSecret }}
+        {{- end }}
+
+      containers:
+        - name: aggregator
+        {{- if .Values.kubecostAggregator.containerSecurityContext }}
+          securityContext:
+            {{- toYaml .Values.kubecostAggregator.containerSecurityContext | nindent 12 }}
+        {{ end }}
+          {{- if .Values.kubecostModel }}
+          {{- if .Values.kubecostModel.openSourceOnly }}
```

Comment:
I also don't see `openSourceOnly` referenced in the values file. What is this? All values need to be accounted for in the values file, even if commented out.

Reply:
added